### PR TITLE
[documentation] Add a nice search experience to the website

### PR DESCRIPTION
**Pull request by s-pace**
This PR will add DocSearch to the documentation website. It will allow an user to have a learn-as-you-type experience by displaying results thanks to a dropdown in a live way.

If there is no result, pressing enter will enable the legacy search.

Ref Bootstraping PR as jquery/api.jquery.com#1104

### CODE REVIEW

**Comment by s-pace** [comment]
We will do an internal check in order to provide you a final PR soon

**Comment by s-pace** [comment]
On iPhone X:
iPhone X

On iPad:
iPad

On desktop:

desktop

[Live demo of this PR
](https://docsearch-jquery-pr.netlify.com/)

Please review it :)

File: themes/jquery/footer.php [hunk 1]
```diff
@@ -37,6 +37,11 @@
 </footer>
 
 <?php wp_footer(); ?>
-
+	<!-- at the end of the BODY -->
+	<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"  onload="docsearch({apiKey: '3cfde9aca378c8aab554d5bf1b23489b',
+	indexName: 'jquery',
+	inputSelector: 'input[name=\'s\']',
+	debug: true // Set debug to true if you want to inspect the dropdown
+	})" async></script>
```
Comment (mgol) [inline themes/jquery/footer.php]: We use tabs for indentation in this file, can you update it? Also, what's inside of the object should be indented more than the closing parens.

File: themes/jquery/css/docsearch.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* Custom DocSeach CSS to adapt the generic one * See https://community.algolia.com/docsearch/styling.html for more info */
+nav#main .searchform {
+    text-shadow: none;
```
Comment (mgol) [inline themes/jquery/css/docsearch.css]: Please use tabs, not spaces.

File: themes/jquery/footer.php (hunk 1 above)
Comment (s-pace) [reply themes/jquery/footer.php]: Sure! Done

**Comment by s-pace** [comment]
Thank you for the guidance @mgol.
Sorry for the .DS_Store, removed ✅ 
Let us know if you need anything else

File: themes/jquery/css/docsearch.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* Custom DocSeach CSS to adapt the generic one * See https://community.algolia.com/docsearch/styling.html for more info */
+nav#main .searchform {
+  text-shadow: none;
```
Comment (mgol) [inline themes/jquery/css/docsearch.css]: This file still uses spaces, please change to tabs.

Comment (s-pace) [reply themes/jquery/css/docsearch.css]: My bad. fixed

**Comment by mgol** [comment]
I've tested it now on https://api.jquery.com/ and it works absolutely _awesome_; such a nice experience. Wonderful work, thanks a lot! 👏

**Comment by dmethvin** [comment]
THIS IS AWESOME! 🎉 Many thanks @s-pace, and @mgol for getting this landed. Much better than the old search.

**Comment by Krinkle** [comment]
@s-pace Hi - is the API key added here part of an account we can access? The QUnit since has since switched to a static site where we push new content with a secret key during CI builds (ref https://github.com/qunitjs/qunit/issues/1460). As I wasn't able to find which account this key belongs to (e.g. to login and generate a secret key), we opted to create an ad-hoc personal account for now and  use. However, I'd rather this not depend on a personal account long-term.

What do you recommend we do?

**Comment by s-pace** [comment]
👋 @Krinkle 
Please [apply to DocSearch from this application](https://docsearch.algolia.com/docs/apply/) form and we will make sure to grant you access to these analytics. The apiKey is not related to any account and it is a search only one that can be safely shared publicly.

**Comment by Krinkle** [comment]
@s-pace I don't think we need/want analytics. I'm referring to the secret key used in jekyll-algolia to push new content after an update. This is currently set jquery/qunit's Jenkins job in a secret `ALGOLIA_API_KEY` environment variable for post-commit builds on the main branch. These are generated from the logged-in Algolia account which is currently a personal one.

Would you recommend we just create another account like that, but with shared credentials stored in a safe place by  OpenJS Foundation? Note that QUnit doesn't use docsearch.js, I don't know if what we use falls under "DocSearch", though.

**Comment by s-pace** [comment]
Thanks for the details. I thought that QUnit was still using DocSearch. I have disabled the daily crawl since this index is not used.

The account you are referring to is tied to @trentmwillis account. I would definitely recommend you to create a shared account with shared credentials stored in a safe place by the OpenJS foundation.

**Pull request merged**